src/esp/gfx/PTexMeshShader.cpp

```diff
@@ -67,13 +67,14 @@ PTexMeshShader::PTexMeshShader() {
              TextureBindingPointIndex::adjFaces);
 
   // cache the uniform locations
-  MVPMatrixUniform_ = uniformLocation("MVP");
-  exposureUniform_ = uniformLocation("exposure");
-  gammaUniform_ = uniformLocation("gamma");
-  saturationUniform_ = uniformLocation("saturation");
-  clipPlaneUniform_ = uniformLocation("clipPlane");
-  tileSizeUniform_ = uniformLocation("tileSize");
-  widthInTilesUniform_ = uniformLocation("widthInTiles");
+  // std::map<std::string, int> uniformLocations;
+  uniformLocations_["MVP"] = uniformLocation("MVP");
+  uniformLocations_["exposure"] = uniformLocation("exposure");
+  uniformLocations_["gamma"] = uniformLocation("gamma");
+  uniformLocations_["saturation"] = uniformLocation("saturation");
+  uniformLocations_["clipPlane"] = uniformLocation("clipPlane");
+  uniformLocations_["tileSize"] = uniformLocation("tileSize");
+  uniformLocations_["widthInTiles"] = uniformLocation("widthInTiles");
```

Comment:
Wait... really? The main point of caching uniform locations was to *reduce* the number of allocations and string operations. This is even worse than calling `uniformLocation()` in every setter. What was wrong with the previous approach?

Reply:
No. With the integer, the sequence of operation is:

1. fetch the ID from memory
2. call `glUniform()` with that ID

With strings, however:

1. call `std::strlen()` on the C char array to get its length
2. allocate memory for a temporary `std::string` of given length
3. copy thta C char array to that allocated memory
4. do a lookup in `std::map` / `std::unordered_map` using the temporary `std::string`
    - either *O(log n)* `std::strcmp()` operations (and lots of memory jumps)
    - or a hash calculation and then at least one `std::strcmp()` operation (and lots of memory jumps)
5. fetch the ID from memory
6. call `glUniform()` with that ID
7. deallocate the temporary string used for key lookup